examples/goldilocks_models/initial_guess.cc

```diff
@@ -0,0 +1,195 @@
+                    VectorXd past_gamma(gamma_dimension);
+                    past_gamma << past_ground_incline(0, 0), past_stride_length(0, 0), past_turning_rate(0, 0);
+                    // normalize gamma
+                    VectorXd normalize_scale = VectorXd::Zero(gamma_dimension);
+                    if(! (min_gi==max_gi))
+                    {
+                        normalize_scale[0] = 1/(max_gi-min_gi);
+                    }
+                    if(! (min_sl==max_sl))
+                    {
+                        normalize_scale[1] = 1/(max_sl-min_sl);
+                    }
+                    if(! (min_tr==max_tr))
+                    {
+                        normalize_scale[2] = 1/(max_tr-min_tr);
+                    }
+                    VectorXd dif_gamma = (past_gamma - current_gamma).array().abs()*normalize_scale.array();
+                    VectorXd dif_gamma2 = dif_gamma.array().pow(2);
+                    double distance_gamma =  (dif_gamma.transpose() * gamma_scale * dif_gamma2)(0,0);
+                    VectorXd w_to_interpolate = readCSV(directory + to_string(past_iter) + string("_")
+                                                        + to_string(sample_num) + string("_w.csv"));
+                    if (distance_gamma == 0) {
+                        w_gamma.conservativeResize(w_to_interpolate.rows(), 1);
+                        w_gamma << w_to_interpolate;
+                        weight_gamma.conservativeResize(1);
+                        weight_gamma << 1;
+                        break;
+                    }
+                    else{
+                        w_gamma.conservativeResize(w_to_interpolate.rows(),w_gamma.cols()+1);
+                        w_gamma.col(w_gamma.cols()-1) = w_to_interpolate;
+                        weight_gamma.conservativeResize(weight_gamma.rows()+1);
+                        weight_gamma(weight_gamma.rows()-1)= 1 /distance_gamma;
+                    }
+                }
+            }
+            double distance_theta =  ((past_theta - current_theta).transpose() * theta_scale *
+                                     (past_theta - current_theta))(0,0);
+            weight_gamma = weight_gamma/weight_gamma.sum();
+            VectorXd w_to_interpolate = w_gamma*weight_gamma;
+            if (distance_theta == 0) {
+                w_theta.conservativeResize(w_to_interpolate.rows(), 1);
+                w_theta << w_to_interpolate;
+                weight_theta.conservativeResize(1);
+                weight_theta << 1;
+                break;
+            }
+            else{
+                w_theta.conservativeResize(w_to_interpolate.rows(),w_theta.cols()+1);
+                w_theta.col(w_theta.cols()-1) = w_to_interpolate;
+                weight_theta.conservativeResize(weight_theta.rows()+1);
+                weight_theta(weight_theta.rows()-1)= 1 /distance_theta;
+            }
+        }
+
+    }
+//    debug for no appropriate theta for interpolation
```

Comment:
Based on my understanding, the only case where we don't get any data is when we set `theta_range` too small.
So normally we should always be able to get past data, and we add drake demand here to prevent the corner case
```
DRAKE_DEMAND(weight_theta.rows()>0)
```

Reply:
Resolved